src/main/java/hudson/plugins/matrix_configuration_parameter/MatrixCombinationsParameterValue.java

```diff
@@ -119,7 +120,7 @@ public boolean equals(Object obj) {
         if (!super.equals(obj)) {
             return false;
         }
-        if (MatrixCombinationsParameterValue.class != obj.getClass()) {
+        if (getClass() != obj.getClass()) {
```

Comment:
<BugInstance type='EQ_GETCLASS_AND_CLASS_CONSTANT' priority='Normal' category='BAD_PRACTICE' message='hudson.plugins.matrix_configuration_parameter.MatrixCombinationsParameterValue.equals(Object) fails for subtypes' lineNumber='122'/>